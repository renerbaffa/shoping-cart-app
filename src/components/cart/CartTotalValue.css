.CartTotalValue-line {
  display: flex;
  justify-content: flex-end;
  margin: 12px 8px;
}

.CartTotalValue-container {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin: 0px 8px;
  min-width: 90px;
}
