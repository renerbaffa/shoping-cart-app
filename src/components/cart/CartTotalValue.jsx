import React from 'react';
import PropTypes from 'prop-types';

import './CartTotalValue.css';

export const CartTotalValue = ({ total }) => (
  <div className="CartTotalValue-line">
    <div className="CartTotalValue-container">
      <b>Total</b>
      <b>$ {total}</b>
    </div>
  </div>
);

CartTotalValue.propTypes = {
  total: PropTypes.number,
};

CartTotalValue.defaultProps = {
  total: 0,
};

export default CartTotalValue;
