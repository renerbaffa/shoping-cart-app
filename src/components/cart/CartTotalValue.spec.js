import React from 'react';
import renderer from 'react-test-renderer';

import { CartTotalValue } from './CartTotalValue';

describe('<CartTotalValue />', () => {
  let component;

  beforeEach(() => {
    component = (<CartTotalValue total={123.3}/>);
  });

  it('should render correctly', () => {
    const cartTotalValue = renderer.create(component).toJSON();
    expect(cartTotalValue).toMatchSnapshot();
  });

  // it('should reach 100% of coverage', () => {
  //   CartTotalValue.defaultProps.onAddProduct();
  // });
});
