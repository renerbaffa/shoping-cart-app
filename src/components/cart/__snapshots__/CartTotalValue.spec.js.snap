// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartTotalValue /> should render correctly 1`] = `
<div
  className="CartTotalValue-line"
>
  <div
    className="CartTotalValue-container"
  >
    <b>
      Total
    </b>
    <b>
      $ 
      123.3
    </b>
  </div>
</div>
`;
